### PR TITLE
add F expression support in queryset.update()

### LINKED ISSUE #245: F Expression for atomic update?
I'm looking for a way to do an atomic update of a field or multiple fields (to increment it)

In Django ORM this is possible using the `F` class, I didn't see anything equivalent in tortoise documentation.

The goal would be to achieve something like this

```
counter = Counter.objects.get_or_create(name = name)
counter.count = F('count') +1
counter.save( update_fields=["count"] )
```

To generate some SQL like

`update counter set count = count + 1`

(In my case I also want to cast in some cases)

`update counter set count = cast(cast(count as integer) + 1) as char)`

Wondering if either of the above are possible yet with tortoise.

### PR DESCRIPTION
**Pull request by long2ice**
## Description

add ``F`` expression support in ``queryset.update()``.
## Motivation and Context

to resolve #245 .
## How Has This Been Tested?

``make test``
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tortoise/queryset.py [hunk 1]
```diff
@@ -658,7 +660,12 @@ def _make_query(self) -> None:
                     db_field = self.model._meta.fields_db_projection[key]
                 except KeyError:
                     raise FieldError(f"Field {key} is virtual and can not be updated")
-                value = executor.column_map[key](value, None)
+                if not isinstance(value, (F, ArithmeticExpression)):
+                    value = executor.column_map[key](value, None)
```
Comment (grigi) [inline tortoise/queryset.py]: Ok, so after investigating, I think due to F and ArithmeticExpression baving different interfaces. Both inherit `pypika.Term`, so maybe just check against that?

File: docs/query.rst [hunk 2]
```diff
@@ -282,3 +282,19 @@ You can view full example here:  :ref:`example_prefetching`
...
+
+For example to use ``F`` to update user balance atomic:
+
+.. code-block:: python3
+
+    from tortoise.expressions import F
+    await User.filter(id=1).update(balance = F('balance') - 10)
+    await User.filter(id=1).update(balance = F('balance') + F('award'), award = 0)
+
+But ``user.balance = F('balance') - 10`` is not supported now.
```
Comment (grigi) [inline docs/query.rst]: I think this last line may confuse users?
May be better to state we ONLY support `F` expressions as part of a `.update()` queryset parameter for now?

Comment (long2ice) [reply docs/query.rst]: ok,I will update docs and fix the error test later.

**Comment by grigi** [comment]
Hi @long2ice Thanks for this work!

**Pull request merged**